src/main/java/domain/user/state/State.java

```diff
@@ -0,0 +1,64 @@
+package domain.user.state;
+
+import domain.card.Card;
+import domain.card.Type;
+import domain.result.GameResult;
+import domain.user.Score;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public abstract class State {
+    protected static final int BLACKJACK_SCORE = 21;
+    private static final int ACE_IS_ELEVEN = 10;
+    private static final int INITIAL_CARDS_SIZE = 2;
+
+    protected List<Card> cards;
+
+    public State(List<Card> cards) {
+        this.cards = new ArrayList<>(cards);
+    }
+
+    public abstract GameResult findResult(State state);
+
+    protected abstract boolean isBlackjack();
+
+    protected abstract boolean isBust();
+
+    protected abstract boolean isStay();
+
+    public abstract State toStay();
+
+    public void add(Card card) {
+        cards.add(card);
+    }
+
+    public Score calculateScore() {
+        int sum = cards.stream()
+                .map(Card::getType)
+                .mapToInt(Type::getScore)
+                .sum();
+        if (sum + ACE_IS_ELEVEN <= BLACKJACK_SCORE && hasAceCard()) {
```

Comment:
```suggestion
        if (hasAceCard() && sum + ACE_IS_ELEVEN <= BLACKJACK_SCORE) {
```
위와 같이 두 논리식의 순서를 변경하는 것이 더 좋을 것 같아요! 🤔 
왜 그런지 고민해보면 좋을 것 같아요 🤗 
혹시라도 이미 이유를 알고 계신다면 코멘트로 간단하게 남겨주시면 좋을 것 같습니다.